common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -157,7 +158,20 @@ class ActiveBulkThread(threading.local):
     """
     def __init__(self, bulk_ops_record_type, **kwargs):
         super(ActiveBulkThread, self).__init__(**kwargs)
-        self.records = defaultdict(bulk_ops_record_type)
+        self._records = defaultdict(bulk_ops_record_type)
+        self.CMS_LEAK_DEBUG_GLOBAL = True  # only log once per process
```

Comment:
I moved this to be an instance var, but since `ActiveBulkThread` inherits from `thread.local` it should still be a singleton per-process.